src/components/space/charts/MemberActionsChart.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+
+import { Typography } from '@mui/material';
+
+import { Action } from '@graasp/sdk';
+
+import { endOfWeek, format } from 'date-fns';
+import { groupBy } from 'lodash';
+import {
+  Bar,
+  CartesianGrid,
+  ComposedChart,
+  Legend,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from 'recharts';
+
+import ChartContainer from '@/components/common/ChartContainer';
+import { getColorForActionTriggerType } from '@/config/constants';
+import { useActionsTranslation, useAnalyticsTranslation } from '@/config/i18n';
+
+type Props = {
+  actionsGroupedByWeekStart: { [key: string]: Action[] };
+};
+const MemberActionsChart = ({
+  actionsGroupedByWeekStart,
+}: Props): JSX.Element => {
+  const { t } = useAnalyticsTranslation();
+  const { t: translateAction } = useActionsTranslation();
+
+  const types: string[] = [];
+
+  const noOfActionTypesOverWeeks = Object.entries(
+    actionsGroupedByWeekStart,
+  ).map(([date, actions]) => {
+    const actionsByType = groupBy(actions, 'type');
+    const typesWithinTheWeek = Object.keys(actionsByType);
+
+    types.push(...typesWithinTheWeek);
+    const result = typesWithinTheWeek.reduce(
+      (acc: { [key: string]: number }, key) => {
+        acc[key] = actionsByType[key].length;
+        return acc;
+      },
+      {},
+    );
+
+    const endOfTheWeek = endOfWeek(date as unknown as Date, {
+      weekStartsOn: 1,
+    }); // Assuming the week starts on Monday
+
+    return {
+      date: `${format(date as unknown as Date, 'MMM dd')} - ${format(endOfTheWeek, 'MMM dd')}`,
```

Comment:
Do not cast, if you fix the type of `actionsGroupedByWeekStart` to Date instead of string, it might work.